### ISSUE TITLE
AC gauges are jumpy on Brief and Overview pages

### ISSUE THREAD

**Issue by blammit** [victronenergy/gui-v2#601]
There's something jumpy about the AC gauges in general, as reported in slack. Needs investigation.

**Comment by blammit** [comment]
Revisit this when issues concerning the gauge calculations in https://github.com/victronenergy/gui-v2/issues/532 are addressed.

Also, some relevant points:
- The max seen value (used as the max value for the AC input gauges) refers to the maximum seen for any AC input, but it should keep a per-input maximum value instead, as the max for one input may be quite different to that for another.
- It's quite possible that Global.acInputs.model is not needed at all, as the Overview and Brief pages only ever use the currently active AC input, and the model is not used in the Device List either.

**Comment by chriadam** [comment]
Might also be an animation thing, see e.g. https://github.com/victronenergy/gui-v2/blob/main/components/controls/Slider.qml#L67

I'm currently working on a shader-based progress arc implementation which should solve a lot of problems, but let's see...

**Comment by chriadam** [comment]
https://github.com/victronenergy/gui-v2/pull/602 still need to reimplement the ProgressArcs and properly test performance on device etc.

**Comment by chriadam** [comment]
Also related to setting appropriate min/max values for gauges, avoiding dynamic scaling.

**Comment by chriadam** [comment]
Marking as low priority, as I don't think this is a showstopper for 1.0.0.

**Comment by kwindrem** [comment]
It's pretty twitchy on a system with minimal load. The AC input gauge swings from max to min with only a 1 or 2 watt change. I'd say something needs to be done for 1.0.0.

It's quite possible that a system with more load produces more stable gauges but have not tried it. Still, it's very distracting with minimal load.

**Comment by chriadam** [comment]
Ah, so this is because of the min/max values not being set properly for the gauges?  Thus, because it's using the "previously seen min / max" values as the rails, it is jumping large amounts visually in reaction to small actual changes?  Or am I misunderstanding?

**Comment by kwindrem** [comment]
Yes, that's my interpretation of what's going on.

**Comment by blammit** [comment]
Yep, let's close this, since #532 addresses these issues. This was originally filed before all of that work was discussed in further detail.

**Comment by chriadam** [comment]
The only other thing which remained was ensuring that we didn't allow two property writes in quick succession to cause an animation jump, IIRC.  See https://github.com/victronenergy/gui-v2/blob/main/components/controls/Slider.qml#L67 for where that was fixed for slider (which we use in the brief page monitor panel data load visualiser IIRC).

**Comment by blammit** [comment]
Ah, I see, https://github.com/victronenergy/gui-v2/pull/602 is still in draft mode and should resolve this? Okay, I'll remove the 1.0.0 tag and leave this open.

**Comment by chriadam** [comment]
Well, #602 is different, not really related to this.  It was an attempt at a performance fix, but proved not to help.

**Comment by blammit** [comment]
Oh, I see, so this is currently fixed for the horizontal gauges / sliders but not the arc gauges?

**Comment by chriadam** [comment]
I believe so.

**Comment by chriadam** [comment]
Closing as wont-fix as the arc gauges have some pretty special logic associated with them to avoid synchronisation issues.
